### ISSUE TITLE
calico-node crashes with empty string value for annotation cni.projectcalico.org/allowedSourcePrefixes

### ISSUE THREAD

**Issue by walshhslaw** [projectcalico/calico#7068]
With `workloadSourceSpoofing: Any`  in the FelixConfiguration adding an annotation with "cni.projectcalico.org/allowedSourcePrefixes"  set to the following in a pod will cause all calico-node pods to go into CrashLoopBackOff:
```
cni.projectcalico.org/allowedSourcePrefixes: '[""]'
```
Tested some IPv6 prefixes too (non exhaustive list), the following causes a calico-node on the specific node where the pod is deployed to no longer have the Ready 1/1 status:
```
cni.projectcalico.org/allowedSourcePrefixes: '["::"]'
cni.projectcalico.org/allowedSourcePrefixes: '["::1"]'
```

## Expected Behavior

Any invalid prefixes would be rejected and not cause calico-node pods to crash

## Current Behavior

For the empty string `cni.projectcalico.org/allowedSourcePrefixes: '[""]'` all calico-node pods go into CrashLoopBackOff with the following error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x147a3c6]

goroutine 201 [running]:
github.com/projectcalico/calico/libcalico-go/lib/backend/syncersv1/updateprocessors.convertWorkloadEndpointV2ToV1Value({0x2ed8aa0?, 0xc0006fcd80})
	/go/src/github.com/projectcalico/calico/libcalico-go/lib/backend/syncersv1/updateprocessors/workloadendpointprocessor.go:166 +0xfa6
github.com/projectcalico/calico/libcalico-go/lib/backend/syncersv1/updateprocessors.(*simpleUpdateProcessor).Process(0xc000c122a0, 0xc000e00900)
	/go/src/github.com/projectcalico/calico/libcalico-go/lib/backend/syncersv1/updateprocessors/processor.go:80 +0x228
github.com/projectcalico/calico/libcalico-go/lib/backend/watchersyncer.(*watcherCache).handleWatchListEvent(0xc0009cc480, 0x347c130?)
	/go/src/github.com/projectcalico/calico/libcalico-go/lib/backend/watchersyncer/watchercache.go:333 +0x57
github.com/projectcalico/calico/libcalico-go/lib/backend/watchersyncer.(*watcherCache).resyncAndCreateWatcher(0xc0009cc480, {0x347c130, 0xc00046a6c0})
	/go/src/github.com/projectcalico/calico/libcalico-go/lib/backend/watchersyncer/watchercache.go:212 +0xa0a
github.com/projectcalico/calico/libcalico-go/lib/backend/watchersyncer.(*watcherCache).run(0xc0009cc480, {0x347c130, 0xc00046a6c0})
	/go/src/github.com/projectcalico/calico/libcalico-go/lib/backend/watchersyncer/watchercache.go:82 +0xa5
github.com/projectcalico/calico/libcalico-go/lib/backend/watchersyncer.(*watcherSyncer).run.func1(0xa7?)
	/go/src/github.com/projectcalico/calico/libcalico-go/lib/backend/watchersyncer/watchersyncer.go:144 +0x65
created by github.com/projectcalico/calico/libcalico-go/lib/backend/watchersyncer.(*watcherSyncer).run
	/go/src/github.com/projectcalico/calico/libcalico-go/lib/backend/watchersyncer/watchersyncer.go:142 +0xc8
```

With the annotation set to `cni.projectcalico.org/allowedSourcePrefixes: '["::1"]'` or `  cni.projectcalico.org/allowedSourcePrefixes: '["::"]'` we see the following error:
```
2022-12-08 01:00:16.423 [PANIC][445437] felix/table.go 1014: Failed to program iptables, giving up after retries error=exit status 2 ipVersion=0x4 table="raw"
panic: (*logrus.Entry) 0xc0001c5180

goroutine 338 [running]:
github.com/sirupsen/logrus.(*Entry).log(0xc0001c50a0, 0x0, {0xc000d90200, 0x33})
	/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:260 +0x47e
github.com/sirupsen/logrus.(*Entry).Log(0xc0001c50a0, 0x0, {0xc000613620?, 0x5?, 0x47004497?})
	/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:304 +0x4f
github.com/sirupsen/logrus.(*Entry).Panic(...)
	/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:342
github.com/projectcalico/calico/felix/iptables.(*Table).Apply(0xc000b4c000)
	/go/src/github.com/projectcalico/calico/felix/iptables/table.go:1014 +0xbc5
github.com/projectcalico/calico/felix/dataplane/linux.(*InternalDataplane).apply.func4(0xc00051c7b8?)
	/go/src/github.com/projectcalico/calico/felix/dataplane/linux/int_dataplane.go:2034 +0x52
created by github.com/projectcalico/calico/felix/dataplane/linux.(*InternalDataplane).apply
	/go/src/github.com/projectcalico/calico/felix/dataplane/linux/int_dataplane.go:2033 +0x717
```
## Possible Solution
For the empty string issue, updating the validation here to catch an empty string:
https://github.com/projectcalico/calico/blob/44452cb395fcc88e5d4ee4e19496aebcff775161/libcalico-go/lib/backend/syncersv1/updateprocessors/workloadendpointprocessor.go#L160-L168

## Steps to Reproduce (for bugs)
1. Set `workloadSourceSpoofing: Any` in default FelixConfiguration 
2. Deploy a pod with metadata.annotations:
    cni.projectcalico.org/allowedSourcePrefixes: '[""]'
3. Monitor state of calico-node pods

## Context
An invalid reference to a value caused a pod to be deployed with `cni.projectcalico.org/allowedSourcePrefixes: '[""]'` causing the cluster to be unusable.

## Your Environment
* Calico 3.24.2
* Kubernetes 1.21.3
* Operating System and version: Ubuntu 20.04

**Comment by caseydavenport** [comment]
Should be a straightforward fix for anyone looking to contribute.

**Comment by caseydavenport** [comment]
The first error should be fixed and show up in v3.25.0 as well as the next v3.24 patch release - v3.24.5 - when it's available.

> With the annotation set to cni.projectcalico.org/allowedSourcePrefixes: '["::1"]' or cni.projectcalico.org/allowedSourcePrefixes: '["::"]' we see the following error:

It's not completely obvious what the error is here, but I suspect it's down to Felix trying to use `::` somewhere that iptables won't accept it, or perhaps in a format or syntax that it's not accepted.

**Comment by caseydavenport** [comment]
Actually, I think the problem there is that `::` is not a valid CIDR - it needs a prefix length. So, that should also be handled by the validation in my PR and I think we can close this.
